libfreerdp/core/transport.c

```diff
@@ -773,6 +838,82 @@ SSIZE_T transport_parse_pdu(rdpTransport* transport, wStream* s, BOOL* incomplet
 			}
 		}
 	}
+	else if (transport->RdstlsMode)
+	{
+		UINT16 version = (header[1] << 8) | header[0];
+		UINT16 pduType;
+		UINT16 dataType;
+
+		if (version != RDSTLS_VERSION_1)
```

Comment:
General remark about `transport_parse_pdu` that goes beyond your changes (but that your changes expose): it really look like we do the job twice.
We have that function that will compute the PDU size, but to do so it partially does the parsing of the PDU. So I guess that we always do the computation of the size that does a partial parsing, and then we do the parsing itself when treating the PDU. One could legitimately ask why not doing all that in a single step: having a smart parsing function that would parse the block and be able to say if there was enough bytes (complete parsing) and how many were expected if not...

Reply:
@jotolo2 did rewrite this to use `wStream` and I´ve pushed 66f657877dd910a611a3007106b0ee04c8842857 that does the `rdpRdstls` parsing in another function